library/private/options_tools.h

```diff
@@ -184,13 +188,114 @@ ratio_t parse(const std::string& str)
 //----------------------------------------------------------------------------
 /**
  * Parse provided string into a color_t.
- * Supported formats: "R,G,B"
- * rely on parse<std::vector<double>>(str)
+ * Supported formats: "R,G,B", "#RRGGBB", "rgb(R,G,B)", "hsl(H,S%,L%)", "hsv(H,S%,V%)",
+ *                    "hwb(H,W%,B%)", cmyk(C%,M%,Y%,K%), "CSS3 color name"
  * Can throw options::parsing_exception in case of failure to parse
  */
 template<>
 color_t parse(const std::string& str)
 {
+  const std::string strCompact = std::regex_replace(str, std::regex("\\s"), "");
+  double rgb[3]{};
+
+  /* Hex format search */
+  const std::regex hexRegex(
+    "^#([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})$", std::regex_constants::icase);
+  std::smatch hexMatch;
+  if (std::regex_search(strCompact, hexMatch, hexRegex))
+  {
+    return color_t(                                  //
+      std::stoul(hexMatch[1], nullptr, 16) / 255.0,  //
+      std::stoul(hexMatch[2], nullptr, 16) / 255.0,  //
+      std::stoul(hexMatch[3], nullptr, 16) / 255.0); //
+  }
+
+  /* RGB format search */
+  const std::regex rgbRegex(
+    "^rgb\\((\\d{1,3}),(\\d{1,3}),(\\d{1,3})\\)$", std::regex_constants::icase);
+  std::smatch rgbMatch;
+  if (std::regex_search(strCompact, rgbMatch, rgbRegex))
+  {
+    rgb[0] = std::stod(rgbMatch[1]) / 255.0;
+    rgb[1] = std::stod(rgbMatch[2]) / 255.0;
+    rgb[2] = std::stod(rgbMatch[3]) / 255.0;
+    if (rgb[0] > 1.0 || rgb[1] > 1.0 || rgb[2] > 1.0)
+    {
+      throw options::parsing_exception("Cannot parse " + str + " into a color_t");
+    }
+    return color_t(rgb[0], rgb[1], rgb[2]);
+  }
+
+  /* Hue-based format search: hsl, hsv, hwb */
+  const std::regex hueRegex(
+    "^([a-z]{3})\\((\\d{1,3}),(\\d{1,3})%{0,1},(\\d{1,3})%{0,1}\\)$", std::regex_constants::icase);
```

Comment:
```suggestion
    "^(hsl|hsv|hwb)\\((\\d{1,3}),(\\d{1,3})%?,(\\d{1,3})%?\\)$", std::regex_constants::icase);
```
you can restrict the known keywords in the regex, and `?` is more idiomatic for "zero or one"

Reply:
will fix that